crates/flowistry_ide/src/mutations/find_mutations.rs

```diff
@@ -0,0 +1,39 @@
+use flowistry::{infoflow::mutation::ModularMutationVisitor, mir::aliases::Aliases};
+use log::debug;
+use rustc_hir::def_id::DefId;
+use rustc_middle::{
+  mir::{visit::Visitor, Body, Location, Place},
+  ty::TyCtxt,
+};
+
+pub fn find_mutations(
+  tcx: TyCtxt<'tcx>,
+  body: &Body<'tcx>,
+  def_id: DefId,
+  place: Place<'tcx>,
+  aliases: Aliases<'tcx>,
+) -> Vec<Location> {
+  let mut locations = vec![];
+  let pointer_aliases = aliases.reachable_values(tcx, body, def_id, place);
+  debug!("pointer aliases: {pointer_aliases:?}");
+
+  ModularMutationVisitor::new(
+    tcx,
+    body,
+    def_id,
+    |mutated_place, _, mutated_location, _| {
+      debug!("checking mutated location {mutated_location:?}");
+
+      let mut place_conflicts = aliases.conflicts(mutated_place).to_owned();
+      place_conflicts.intersect(&pointer_aliases);
+
+      if place_conflicts.len() > 0 {
+        debug!("  found conflicts: {place_conflicts:?}");
+        locations.push(mutated_location);
+      }
+    },
+  )
+  .visit_body(body);
+
+  locations
```

Comment:
Do you think it's clearer to just write `locations` or to end with `return locations;`? I've never had a clear policy on this.

Reply:
When writing this it seemed fine to use `locations` (and I had just started picking up Rust and wanted to use fun new syntax) but reading it again a while afterwards makes me prefer `return locations;` for clarity. I think omitting `return` is better for shorter closures.